src/Resources/web/app_dev.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+use Contao\ManagerBundle\ContaoManager\Plugin as ManagerBundlePlugin;
+use Contao\ManagerBundle\HttpKernel\ContaoKernel;
+use Doctrine\Common\Annotations\AnnotationRegistry;
+use Symfony\Component\Debug\Debug;
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\HttpFoundation\Request;
+
+// The check is to ensure we don't use .env in production
```

Comment:
@aschempp I have to second the statement by @backbone87.

The original line from symfony (`if (!getenv('APP_ENV')) {`) has absolutely nothing in common with your `file_exists` check here.
The whole workflow is changed.
IMO we should keep the behaviour of symfony and therefore add the `dotenv` to both endpoints.

> Also, as you pointed out, we don't need anything from .env in prod, so no reason to read and parse a unused file.

You mean, your endpoint does not need anything so far but can you guarantee that no bundle does some `getenv` because ist is already `dotenv` aware?
What do you expect users to do when they want to activate it? I mean, the can not alter the endpoint at all as the cm will overwrite it over and over again.